bin/Url.py

```diff
@@ -16,6 +18,13 @@
 
 from Helper import Process
 
+# Used to prevent concat with empty fields due to url parsing
+def avoidNone(str):
+    if str is None:
```

Comment:
It's generally recommended to avoid using [reserved word](http://stackoverflow.com/questions/22864221/is-the-list-of-python-reserved-words-and-builtins-available-in-a-library) (such as `str`)